### PR TITLE
Add network-get support to ModelBackend

### PR DESCRIPTION
**Pull request by dshcherb**
Hook tool handling for network-get which may be a base for further
endpoint modeling in the model.

### CODE REVIEW

File: test/test_model.py [hunk 1]
```diff
@@ -767,6 +768,40 @@ def test_storage_tool_errors(self):
...
+        self.assertEqual(network_info, json.loads(network_get_out))
+        self.assertEqual(fake_script_calls(self, clear=True), [['network-get', 'deadbeef', '--format=json']])
+
+        network_info = self.backend.network_get('deadbeef', 1)
+        self.assertEqual(network_info, json.loads(network_get_out))
+        self.assertEqual(fake_script_calls(self, clear=True), [['network-get', 'deadbeef', '-r', '1', '--format=json']])
+
+    def test_network_get_errors(self):
+        err_no_endpoint = "ERROR no network config found for binding \"$2\""
+        err_no_rel = "ERROR invalid value \"$3\" for option -r: relation not found"
```
Comment (chipaca) [inline test/test_model.py]: I wonder about this (and similar) lines. As soon as I need to backslash a double quote, in Python, I'd change the type of the outer quotes... that is, in this case, I think I'd write this like
```suggestion
        err_no_rel = 'ERROR invalid value "$3" for option -r: relation not found'
```

above you use triple-single-quotes and then inside use double and single, so it doesn't seem to be stylistic.

Anyway, just wondering about it.

File: ops/model.py [hunk 2]
```diff
@@ -638,3 +638,19 @@ def function_log(self, message):
...
+        relation_id -- An optional relation id to get network info for.
+        """
+        cmd = ['network-get', endpoint_name]
+        if relation_id is not None:
+            cmd.extend(['-r', str(relation_id)])
+        try:
+            return self._run(*cmd, return_output=True, use_json=True)
+        except ModelError as e:
+            if 'relation not found' in str(e):
+                raise RelationNotFoundError() from e
```
Comment (niemeyer) [inline ops/model.py]: Digging into error messages is a bad idea as we discussed in similar cases before.

Why is that necessary?

Comment (dshcherb) [reply ops/model.py]: In this case I am following what we agreed on in #60 for other hook tools:

https://github.com/canonical/operator/pull/60#pullrequestreview-324357752
https://github.com/canonical/operator/blame/master/ops/model.py#L536-L564

IIRC, this was a compromise as Juju does not provide a better way to indicate a particular error type.

> Why is that necessary?

To multiplex based on what happened at the controller side:

1. a generic failure at the controller side;
2. a relation with the specified ID does not exist.

Comment (niemeyer) [reply ops/model.py]: Sorry, my bad. I forgot that this was a case we already agreed on.

File: test/test_model.py [hunk 3]
```diff
@@ -767,6 +768,40 @@ def test_storage_tool_errors(self):
                 run()
             self.assertEqual(fake_script_calls(self, clear=True), calls)
 
+    def test_network_get(self):
+        network_get_out = ('{"bind-addresses":[{"mac-address":"","interface-name":"",'
+                           '"addresses":[{"hostname":"","value":"192.0.2.2","cidr":""}]}]'
+                           ',"egress-subnets":["192.0.2.2/32"],"ingress-addresses":["192.0.2.2"]}')
```
Comment (niemeyer) [inline test/test_model.py]: Not the usual quoting here and elsewhere in the PR, plus point by @chipaca is still open
